### ISSUE TITLE
Automate part of accessibility-insights

### ISSUE THREAD

**Issue by kamal2222ahmed** [microsoft/accessibility-insights-web#6172]
## It takes too long to test a complete WebSite for Accessibility, especially since:
1. Current WebSites that I am working on are content rich
2. They retain historical data
3. It's basically Driven by Drupal

## Describe the desired outcome
I would like at least 50% - 70% of the 508 compliance be automatically conducted. This must follow the same test steps/results as a person would do using Trusted Tester V5.0 Guidelines ( Success Criteria ) Which is the same as WCAG2.1 AA and A  

## Describe alternatives you've considered
I tried just using FastPass, but:

1. It does not indicate what percentage of the page was scanned
2. I would have to repeat the same test for thousands of pages
3. With almost all websites now JavaScript Driven, so the pages are generated only when the user uses them, so the total number of pages could even be higher than assumed
4. No provision to crawl the website and run the scan on multiple pages

## Additional context
Would be nice to run it from command line, so we can make it part of the CI/CD Pipeline
Plus, results can be published and compared over time, to do regression analysis

**Comment by dbjorge** [comment]
Hi @kamal2222ahmed,

It looks like there are several different requests bundled into one issue here. I'll respond one by one:

> I would like at least 50% - 70% of the 508 compliance be automatically conducted.

We would love this, too! Unfortunately, we don't think this is currently feasible - we are constantly reevaluating and thinking about how to automate more types of accessibility checks, but the unfortunate reality is that there are many types of accessibility issue which cannot be detected by automation. Many types of issues fundamentally boil down to "there is no way programmatic way for an accessibility technology to identify *&lt;some thing on a page>*" - but for cases like that, automated accessibility checkers like ours often have exactly the same problem detecting the problematic components that an accessibility technology would! If there was a programmatic way for accessibility checkers to detect such cases, accessibility technologies would just use that.

If you have more specific suggestions for specific types of issues that you can propose an automated means of detecting, we would be happy to evaluate them. It would be best to file separate, individual issues for such cases.

> 1. It does not indicate what percentage of the page was scanned

I'm not sure I understand what you mean by this. FastPass and Assessment both involve scanning the entire page in question (in its current state).

If you're asking for a percentage like "what percentage of potential issues that might exist on the page did automated checks check for" or "how many WCAG criteria did automated checks cover" or similar: We prefer to avoid giving specific percentages for this, for several reasons:

* There are many WCAG SCs which are only partially covered by automated checks
* There is a lot of variation between different sites/pages in how many accessibility issues exist that can be detected with automation, and the automation can't know how many issues might exist on a given page that automated checks can't catch
* Even for a specific given page, different experts will disagree about whether repeated instances of a similar issue should count as separate issues or not for the sake of calculating percentages like this

> 2. I would have to repeat the same test for thousands of pages
> 3. With almost all websites now JavaScript Driven, so the pages are generated only when the user uses them, so the total number of pages could even be higher than assumed
> 4. No provision to crawl the website and run the scan on multiple pages
>
> Would be nice to run it from command line, so we can make it part of the CI/CD Pipeline.
> Plus, results can be published and compared over time, to do regression analysis

We actually already have a command line solution for this which can crawl a whole site and perform our automated checks against each page! It is published as the [accessibility-insights-scan NPM package](https://www.npmjs.com/package/accessibility-insights-scan).

We also maintain an [Accessibility Insights for Azure DevOps extension](https://github.com/microsoft/accessibility-insights-action/blob/main/docs/ado-extension-usage.md) which provides similar functionality as an Azure DevOps Pipelines task and an [Accessibility Insights Action](https://github.com/microsoft/accessibility-insights-action/blob/main/docs/gh-action-usage.md) for GitHub Actions.

Finally, if you already have an existing set of end-to-end tests for your site, we also maintain a set of [sample code](https://github.com/microsoft/axe-pipelines-samples) which has example of how to integrate the accessibility scanning engine we use ([axe-core](https://github.com/dequelabs/axe-core)) into test suites that use a few popular browser automation tools like [Playwright](https://playwright.dev/) and [Selenium](https://www.selenium.dev/).

@kamal2222ahmed , do these options cover what you were looking for regarding multi-page crawling?

@ferBonnin / @peterdur , I think a good takeaway for our team from this issue is that it might be helpful for us to include cross-links to these options from within Accessibility Insights for Web itself (and/or from https://accessibilityinsights.io)
